src/Components/Topic.jsx

```diff
@@ -0,0 +1,90 @@
+import { useEffect, useState } from "react";
+import { Link, useParams } from "react-router-dom";
+import { getTopics, getArticles } from "./api";
+import ArticleList from "./ArticleList";
+import "./Topic.css";
+
+export default function Topic() {
+  const { topic: initialTopic } = useParams();
+  const [topics, setTopics] = useState([]);
+  const [currentTopic, setCurrentTopic] = useState(initialTopic || "");
```

Comment:
Is this state necessary? The current topic will always be found in the value from `useParams`